logstash-core/spec/logstash/plugins/builtin/pipeline_input_output_spec.rb

```diff
@@ -101,6 +101,27 @@ def stop_input
         output.do_close
       end
     end
+
+    it "stopped input should process events until upstream outputs stop" do
+      start_input
+      output.register
+      pipeline_bus.setBlockOnUnlisten(true)
+
+      output.multi_receive([event])
+      expect(queue.pop(true).to_hash_with_metadata).to match(event.to_hash_with_metadata)
+
+      @thread = Thread.new do
```

Comment:
this instance variable holding the thread object is never acted upon, or am I missing something?
Also, this is the equivalent of just `Thread.new { input.do_stop }`